WalletWasabi.Tests/RegTests.cs

```diff
@@ -1135,6 +1136,133 @@ private void WalletTestsAsync_MemPoolService_TransactionReceived(object sender,
+
+			try
+			{
+				nodes.Connect(); // Start connection service.
+				node.VersionHandshake(); // Start mempool service.
+				indexDownloader.Synchronize(requestInterval: TimeSpan.FromSeconds(3)); // Start index downloader service.
+
+				// Wait until the filter our previous transaction is present.
+				var blockCount = await Global.RpcClient.GetBlockCountAsync();
+				await WaitForFiltersToBeProcessedAsync(TimeSpan.FromSeconds(120), blockCount);
+
+				using (var cts = new CancellationTokenSource(TimeSpan.FromSeconds(30)))
+				{
+					await wallet.InitializeAsync(cts.Token); // Initialize wallet service.
+				}
+				Assert.Single(wallet.Coins);
+
+				// Send money before reorg.
+				var operations = new[]{
+					new WalletService.Operation(scp, Money.Coins(0.5m), "") };
+				var btx = await wallet.BuildTransactionAsync("password", operations, 2);
+
+				operations = new[]{
+					new WalletService.Operation(scp, Money.Coins(0.5m), "") };
+				btx = await wallet.BuildTransactionAsync("password", operations, 2);
+
+				// Test synchronization after fork.
+				// Invalidate the blocks containing the funding transaction
+				var tip = await Global.RpcClient.GetBestBlockHashAsync();
+				await Global.RpcClient.InvalidateBlockAsync(tip); // Reorg 1
+				tip = await Global.RpcClient.GetBestBlockHashAsync();
+				await Global.RpcClient.InvalidateBlockAsync(tip); // Reorg 2
+
+				// Generate two new blocks (replace the previous invalidated ones)
+				_filtersProcessedByWalletCount = 0;
+				await Global.RpcClient.GenerateAsync(3);
+				await WaitForFiltersToBeProcessedAsync(TimeSpan.FromSeconds(120), 3);
+
+				// Send money after reorg.
+				// When we invalidate a block, those transactions setted in the invalidated block
+				// are reintroduced when we generate a new block though the rpc call 
+				operations = new[]{
+					new WalletService.Operation(scp, Money.Coins(0.5m), "") };
+				btx = await wallet.BuildTransactionAsync("password", operations, 2);
+
+				operations = new[]{
+					new WalletService.Operation(scp, Money.Coins(0.5m), "") };
+				btx = await wallet.BuildTransactionAsync("password", operations, 2);
+
+				
+				// Test synchronization after fork with different transactions.
+				// Create a fork that invalidates the blocks containing the funding transaction
+				_filtersProcessedByWalletCount = 0;
+				var winningFork = await RegTestFixture.BackendRegTestNode.GenerateEmptyBlocksAsync(100,
+					new Key().PubKey.GetAddress(Global.RpcClient.Network), 10);
+
+				tip = await Global.RpcClient.GetBestBlockHashAsync();
+				Assert.Equal(tip, winningFork.Last().GetHash());
+				
+				await WaitForFiltersToBeProcessedAsync(TimeSpan.FromSeconds(120), 10);
+
+				// When we creates a winning fork, those transactions setted in the orphand blocks
+				// do not exist any more
+				var coin = wallet.Coins.First();
+				//var txCoin = await Global.RpcClient.GetRawTransactionAsync(coin.TransactionId, true);
```

Comment:
The problem with Nicolas's GetRawTransaction is that it gives us a transaction object, while Core's RPC gives us more information, like if the coin is confirmed or not. That's why I used the SendCommand instead above in my PR comment.